File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hash_map = {}
        
        for num in nums:
            if num in hash_map:
                hash_map[num] += 1
            else:
                hash_map[num] = 1
        
        lst = sorted(list(hash_map.items()), key=lambda x: x[1], reverse=True)
        sliced = lst[slice(k)]
        
        return list(map(lambda x: x[0], sliced))

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */

var addToMap = function(map, item) {
    if (map.has(item)) {
        const curVal = map.get(item)
        map.set(item, curVal + 1)
    } else {
        map.set(item, 1)
    }
}

var topKFrequent = function(nums, k) {
    let counter = new Map()
    
    for (num of nums) {
        addToMap(counter, num)
    }
    
    let keys = Array.from(counter.keys())
    // sorts the keys of the counter map by their values in descending order
    keys.sort((a, b) => counter.get(b) - counter.get(a))
    
    return keys.slice(0, k)
};